lib/ansible/modules/cloud/amazon/iam_oidc_federation.py

```diff
@@ -0,0 +1,365 @@
+                                        url
+                                    ),
+                                )
+            res["oidc_provider"] = self._build_res(arn)
+
+        # The IdP does not exist; create it.
+        else:
+            res["changed"] = True
+            if not self.module.check_mode:
+                try:
+                    resp = self._create_open_id_connect_provider(
+                        url, client_ids, thumbprints
+                    )
+                    res["oidc_provider"] = self._build_res(
+                        resp["OpenIDConnectProviderArn"]
+                    )
+                except botocore.exceptions.ClientError as e:
+                    self.module.fail_json_aws(
+                        e,
+                        msg="Could not create the OIDC identity provider '{0}'".format(
+                            url
+                        ),
+                    )
+
+        self.module.exit_json(**res)
+
+    def delete_oidc_provider(self, url):
+        res = {"changed": False}
+        try:
+            arn = self._get_provider_arn(url)
+        except (
+            botocore.exceptions.ValidationError,
+            botocore.exceptions.ClientError,
+        ) as e:
+            self.module.fail_json_aws(
+                e,
+                msg="Could not get the ARN of the OIDC identity provider '{0}'".format(
+                    url
+                ),
+            )
+
+        # If OIDC provider exists; delete it.
+        if arn:
+            res["changed"] = True
+            if not self.module.check_mode:
+                try:
+                    self._delete_open_id_connect_provider(arn)
+                except botocore.exceptions.ClientError as e:
+                    self.module.fail_json_aws(
+                        e,
+                        msg="Could not delete the OIDC identity provider '{0}'".format(
+                            url
+                        ),
+                    )
+
+        self.module.exit_json(**res)
+
+    def _build_res(self, arn):
+        oidc_provider = self._get_open_id_connect_provider(arn)
+        return {
+            "arn": arn,
+            "url": oidc_provider["Url"],
+            "client_id_list": oidc_provider["ClientIDList"],
+            "thumbprint_list": oidc_provider["ThumbprintList"],
+            "create_date": oidc_provider["CreateDate"].isoformat(),
+        }
+
+
+def main():
+    argument_spec = dict(
+        url=dict(required=True, aliases=["name"]),
+        client_ids=dict(
+            required=False, type="list", elements="str", aliases=["audiences"],
+        ),
+        thumbprints=dict(required=False, type="list", elements="str"),
+        state=dict(default="present", choices=["present", "absent"],),
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[("state", "present", ["client_ids", "thumbprints"])],
```

Comment:
I'm only familiar with SAML federation rather than OIDC federation...

Why require both of these?
Would it be possible (or make sense) to set the client_ids and thumbprints independently or are they linked?

Reply:
Thumbprints are required by the API during OIDC provider creation. client_ids could potentially be empty, but I'm not sure if there'd be any utility there. Will reconsider.